### ISSUE TITLE
Question about the warning option for the clang

### ISSUE THREAD

**Issue by i80287** [llvm/llvm-project#114125]
Compilation of the following code produces 2 warnings about potential null pointer dereference and null pointer dereference when compiled by the gcc with flags `-O2 -Wnull-dereference`.
This code contains ub and compiler is not obligated to warn about it, but is it possible to add additional warning option to the clang so that it reports about possible null pointer dereference like gcc does?
```cpp
#include <cstdint>
#include <vector>

auto f(std::uint32_t n) {
    std::vector<std::int32_t> v(n + 1);
    v[0] = 1;
    return v;
}

int main() {
    f(static_cast<std::uint32_t>(-1));
}
```
On the godbolt
https://godbolt.org/z/4q4bqcx5G

**Comment by i80287** [comment]
Maybe there exists some option to warn about potential null dereference (even if false positive if possible)?
I've tried this set of options, but got no success:
`-O2 -Wall -Wextra -Wcast-qual -Wpedantic -Wunused -Wshadow -Wnull-dereference -Wundef -Wsign-conversion -Wsign-compare -Wconversion -Wmissing-noreturn -Wunreachable-code -Wcast-align -Wformat=2 -Wswitch-bool -Wswitch-default -Wswitch-enum -Wdeprecated -Wtype-limits -Wshift-negative-value -Walloca -Werror -Weffc++ -Wsign-promo -Wold-style-cast -Warray-bounds -Warray-bounds-pointer-arithmetic -Wshift-overflow -Wshift-sign-overflow -Wshorten-64-to-32 -Wthread-safety -Wnull-pointer-arithmetic -Wnullable-to-nonnull-conversion -Wsometimes-uninitialized -Wstatic-in-inline -Wformat-pedantic -Wformat-type-confusion -Wfour-char-constants -Wgcc-compat -Wgnu -Widiomatic-parentheses -Wimplicit -Wimplicit-fallthrough -Winconsistent-missing-destructor-override -Wframe-address -Wnull-pointer-subtraction -Warray-parameter -Wincompatible-function-pointer-types-strict -Wsigned-enum-bitfield`

**Comment by shafik** [comment]
Looks similar to: https://github.com/llvm/llvm-project/issues/113068

**Comment by i80287** [comment]
> Looks similar to: #113068

Yes, seems it's the same missing warning as in this issue
Ig then this issue should be marked as duplicate and closed?
